File: rust_with_atcoder/atcoder_practice/abc162_c/another_langs/162c.php
<?php

$k = intval(trim(fgets(STDIN)));
$ans_sum = 0;

for ($a = 1; $a <= $k; $a++) {
	for ($b = 1; $b <= $k; $b++) {
		for ($c = 1; $c <= $k; $c++) {
			$ans_sum += gcd(gcd($a, $b), $c);
		}
	}
}

echo $ans_sum . PHP_EOL;

function gcd(int $a, int $b): int {
	if ($b === 0) {
		return $a;
	}
	return gcd($b, $a % $b);
}

File: rust_with_atcoder/atcoder_practice/abc162_c/another_langs/162c.py
import sys


def gcd(a: int, b: int) -> int:
	if b == 0:
		return a
	return gcd(b, a % b)


def main() -> None:
	k = int(sys.stdin.readline().strip())
	ans_sum = 0

	for a in range(1, k + 1):
		for b in range(1, k + 1):
			for c in range(1, k + 1):
				ans_sum += gcd(gcd(a, b), c)

	print(ans_sum)


if __name__ == "__main__":
	main()

